src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.cs

```diff
@@ -483,6 +483,41 @@ private static void StopThread(Thread thread)
             }
         }
 
+
+        private static int StackSizeFromConfig
+        {
+            get => _stackSizeFromConfig ??= GetDefaultStackSize();
+        }
+
+        private static int? _stackSizeFromConfig;
+
+        internal static int GetDefaultStackSize()
+        {
+            // Keep the same arbitrary minimum and maximum from the coreclr\vm layer.
+            const uint minStack = 0x10000;     // 64K
+            const uint maxStack = 0x80000000;  //  2G
+
+            const string stackSizeVar = "DefaultStackSize";
+            const string stackSizeEnvVar = "DOTNET_DefaultStackSize";
```

Comment:
This shortcut (making the AppContext switch name same as the environment variable name) makes things inconsistent with how we normally name AppContext switches. See e.g. [documented GC switches](https://learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector) where `DOTNET_gcServer` environment variable is a synonym for `System.GC.Server` host config option (i.e. these are not prefixes and AppContext switches adhere to a naming convention).

It is a bit more work to do this correctly. Sample how it's done on the managed side (also notice the naming convention):

https://github.com/dotnet/runtime/blob/8fca0a156dcbb1eba5c79e387a6d8f924bfdd3a6/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs#L66-L70

On the native side we do this (it's what the GC uses to read both env variable settings and host config options):

https://github.com/dotnet/runtime/blob/8fca0a156dcbb1eba5c79e387a6d8f924bfdd3a6/src/coreclr/nativeaot/Runtime/gcenv.ee.cpp#L770-L784

Reply:
Also, it would be nice to keep the configuration capabilities between NAOT and regular CoreCLR the same - make the configuration switch work the same way for both NAOT and CoreCLR.